app/src/main/java/com/concordia/mcga/helperClasses/DatabaseHelper.java

```diff
@@ -0,0 +1,147 @@
+package com.concordia.mcga.helperClasses;
+
+import android.content.Context;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteException;
+import android.database.sqlite.SQLiteOpenHelper;
+import android.util.Log;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.sql.SQLException;
+
+/**
+ * Part of the code was taken from https://blog.reigndesign.com/blog/using-your-own-sqlite-database-in-android-applications/
+ */
+
+public class DatabaseHelper extends SQLiteOpenHelper {
+    private static final String DATABASE_NAME = "development.db";
+    public static SQLiteDatabase db;
+    //The Android's default system path of our application database.
+    private static String databasePath;
+    private static Context context;
+    private static DatabaseHelper databaseHelper;
+
+
+    private DatabaseHelper(Context context) {
+        super(context, DATABASE_NAME, null, 1);
+        DatabaseHelper.context = context;
+    }
+
+    public static boolean setupDatabase(Context context) {
+        if (db == null) {
+            DatabaseHelper.context = context;
+            databasePath = context.getFilesDir().getPath() + "/../databases/";
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public static DatabaseHelper getInstance() {
+        if (context == null) {
+            return null;
+        }
+        if (db == null) {
+            return databaseHelper = new DatabaseHelper(context);
+        } else {
+            return databaseHelper;
+        }
+
+    }
+
+    /**
+     * Creates a empty database on the system and rewrites it with your own database.
+     */
+    public void createDataBase() throws IOException {
+        boolean dbExist = checkDataBase();
```

Comment:
Programming wise, I agree it is unnecessary but it makes the code easier to understand. If(!checkDataBase) doesn't make much sense to me. It's just personal choice.